src/utils/hooks/useDataViews.tsx

```diff
@@ -55,49 +56,54 @@ export const useDataViewsList = () => {
   return { error, loading, dataViewsList }
 }
 
-export const useDataViewsTable = ({ dataViewCode, apiQueries, filter }: DataViewTableProps) => {
+export const useDataViewFilterDefinitions = (dataViewCode: string) => {
   const { strings } = useLanguageProvider()
+  const [filterDefinitions, setFilterDefinitions] = useState<FilterDefinitions>()
   const [error, setError] = useState<ErrorResponse | null>(null)
-  const [loading, setLoading] = useState(false)
+  const [loading, setLoading] = useState(true)
+
+  useEffect(() => {
+    processRequest({
+      url: getServerUrl('dataViews', { dataViewCode, query: { first: '0' } }),
+      setError,
+      setLoading,
+      setStateMethod: (response: DataViewsTableResponse) =>
+        setFilterDefinitions(buildFilterDefinitions(response, strings.DATA_VIEW_NULL_VALUE)),
+      filter: {},
+    })
+  }, [dataViewCode])
+
+  return { filterDefinitions, loading, error }
+}
+
+export const useDataViewsTable = ({
+  dataViewCode,
+  apiQueries,
+  filter,
+  filtersReady,
+}: DataViewTableProps) => {
+  const [error, setError] = useState<ErrorResponse | null>(null)
+  const [loading, setLoading] = useState(true)
   const [dataViewTable, setDataViewTable] = useState<DataViewsTableResponse>()
-  const [filterDefinitions, setFilterDefinitions] = useState<any>()
   const {
     userState: { templatePermissions },
   } = useUserState()
 
-  const updateDataViewTable = (newDataViewTable: DataViewsTableResponse) => {
-    // We only want to load filterDefinitions once (per dataView code),
-    // otherwise it gets into an infinite loop. Similarly, we don't want to set
-    // the dataViewTable data until we've loaded filterDefinitions, or the Table
-    // page will display the full list briefly before showing the filtered list.
-
-    // TO-DO: Fetch filter definitions in a separate query *before* requesting
-    // dataViewTable
-
```

Comment:
As mentioned earlier, all the changes in this file solve the "To-Do" mentioned here.

Reply:
Separating into two hooks makes it a lot simpler, and runs much nicer.